### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by medic-code** [eduardoleao052/js-pytorch#23]
@eduardoleao052 Just curious as to where you see the project going in the short term, medium term and long term. Realise this has mostly been a one person endeavour and you won't have lots of things figured out but thought I would pose the question and whether its worth thinking of a quick roadmap. You've mentioned that you want it to be accessible and low barrier for JS learners but would good to see your understanding of it as it stands now and what the future direction might look like.

See https://github.com/KnowledgeCanvas/knowledge/wiki/Feature-Roadmap - for examples.  Somethings relatively quick and easier to do.

You can also consider doing a github projects to help gather your thoughts.

**Comment by eduardoleao052** [comment]
@medic-code 
That's super cool! Many of these tools you're pointing out I didn't even know lol
I'll take a look.

But to summarize, at least mid-term, I see this project as being able to fill a (not so small) niche of JS developers that are interested in AI (maybe have been following the GEN-AI hype, Devin and such) but do not have Python knowledge. So if these guys want to work with decent ML tools and run say demos or small apps on the web they can do it with this project.
I don't see it as a way to "replace" PyTorch and the bigger players, but it still could have a good role to play.

On another topic, did you see the new issue templates? I tried to add some more fields to the bug report, and kept the feature suggestion mostly the same. Is the result close to what you were imagining?

**Comment by medic-code** [comment]
langchain-js's community is larger than you think for example! I would maybe start to break that down knowing your objective and what you think we should and should not implement going forward. 

Few comments, 

1. On the bug report it seems pedantic but instead of a small snippet of code, I would say minimal reproducible code that exemplifies the bug. 
1. 1 Sometimes worth providing a link for those who don't have a clue what this means - https://support.gurobi.com/hc/en-us/articles/13941323633681-Tutorial-Preparing-a-Minimal-Reproducible-Example - for example

2. Consider adding a category for actual behaviour next to expected behaviour - you could somewhat argue you provide this information in the description but it can be quite nice to see actual and expected behaviour next to each other concisely. 

3. Add in optional screenshot section (or screengrab)- I try always to provide screen grabs in cases where the description in words is too verbose. 

4. Add in a section for environment - Sometimes this can be important in teasing out bugs that are to do with the system
4.1 Operating system
4.2 Version of the repo or branch 
4.3 Version of node.js 
4.4 If applicable browser name and version 
5. A check list for a bug report something like https://github.com/langchain-ai/langchainjs/issues/new?assignees=&labels=02+Bug+Report&projects=&template=bug-report.yml

For the feature request, maybe provide a check list like https://github.com/langchain-ai/langchainjs/discussions/4862

**Comment by eduardoleao052** [comment]
@medic-code 
I think I expressed myself poorly, sorry! I've heard of `langchain.js`, the tools I wasn't really familiar with that I mentioned were building Issue Templates and Project Roadmaps.

By the way, I updated the Issue templates, if you want to take a look. Now they have input fields, and seem to be a lot better organized. Plus, I implemented the changes you suggested. If you think it could still be improved, let me know!
